### ISSUE TITLE
cran.md reporting 0 packages from CRAN

### ISSUE THREAD

**Issue by eliocamp** [r-lib/revdepcheck#342]
I'm running revdepcheck and after seemingly checking all packages correctly, the file cran.md states:

```
## revdepcheck results

We checked 36 reverse dependencies (0 from CRAN + 36 from Bioconductor), comparing R CMD check results across CRAN and dev versions of this package.

 * We saw 0 new problems
 * We failed to check 0 packages

```

However I'm pretty certain that most if not all the packages I checked are on CRAN. The package is [ggnewscale](https://cran.r-project.org/package=ggnewscale). 

For example, one of the reverse dependencies is [cmcR](https://cran.r-project.org/web/packages/cmcR/index.html), which is on CRAN and not in Bioconductor.

**Comment by gaborcsardi** [comment]
I think you are using RSPM, and repdevcheck does not know that that is CRAN. It could try to guess it. What's the output of
```r
getOption("repos")
```

**Comment by eliocamp** [comment]
Yes, I'm using RSPM: "https://packagemanager.rstudio.com/all/__linux__/focal/latest"

**Comment by gaborcsardi** [comment]
Can you please run 
```
getOption("repos")
```
and show the complete output? Details might matter. Thanks!

**Comment by eliocamp** [comment]
Yep, no problem. 

```
> getOption("repos")
                                                           RSPM 
"https://packagemanager.rstudio.com/all/__linux__/focal/latest
```

**Comment by gaborcsardi** [comment]
Yeah, so if you name that repo `CRAN` instead of `RSPM`, then revdepcheck will _probably_ know that they are CRAN packages. (I haven't tested, though.)

**Comment by eliocamp** [comment]
I've just renamed the repo and ran it again. I still get the same message.

**Comment by gavinsimpson** [comment]
I'm also using RSPM. I see the same behaviour if the revdep check process times out as per #355; solution there is to make `timeout` a very large number (it's in seconds). Increasing the `timeout` allows the revdep checks to complete, but do not the second comment in #355 as this timeout applies to the entire revdep checking process (at the time of writing) and if your package's reverse dependencies themselves have a lot of dependencies, just installing those dependencies can take a very long time indeed

**Comment by MichaelChirico** [comment]
I think I'm seeing a similar problem on `revdep_check()` of {bit64}.

What I notice is that my `revdep_check()` terminated mid-way for an unknown reason; `README.md` correctly reports `## Failed to check (85)`, but `cran.md` incorrectly reports `(17 from CRAN + 85 from Bioconductor)` (glancing at the CRAN landing page we see ~8 Bioconductor revdeps: https://cran.r-project.org/web/packages/bit64/index.html).

It seems all too convenient that those two numbers match exactly, so I suspect `revdep_check()` has somehow mixed up "these packages failed" to mean "they must come from Bioconductor".

**Comment by MichaelChirico** [comment]
Based on this that would make sense:

https://github.com/r-lib/revdepcheck/blob/5a0c13819b83567a4c789311131d69dfa35a722c/R/report.R#L294
https://github.com/r-lib/revdepcheck/blob/5a0c13819b83567a4c789311131d69dfa35a722c/R/report.R#L284
https://github.com/r-lib/revdepcheck/blob/5a0c13819b83567a4c789311131d69dfa35a722c/R/report.R#L330-L332

We're categorizing Bioconductor :left_right_arrow: `!isTRUE(<cran flag set>)`, but "the check failed" is another reason that flag wouldn't be set.
